File: assignment-4-Jwizzed-main/Authentication.py
from Admin import Admin
from AdminRecordManagement import AdminRecordManagement
from Student import Student
from StudentRecordManagement import StudentRecordManagement


class Authentication:
    def __init__(self):
        pass

    def register(self, name, idNumber, contactNumber, role):

        if role == "RA":
            admin = Admin(name, idNumber, contactNumber)
            admin_record_management = AdminRecordManagement()
            admin_record_management.add(admin)

        elif role == "RS":
            student = Student(name, idNumber, contactNumber)
            student_record_management = StudentRecordManagement()
            student_record_management.add(student)
        else:
            raise ValueError("User not found\n")

    def login(self, name, idNumber, contactNumber, role):

        if role == "LA":
            admin = Admin(name, idNumber, contactNumber)
            admin_record_management = AdminRecordManagement()
            found = admin_record_management.findRecord(admin.name, admin.idNumber,admin.contactNumber)
            if found:
                return True
            else:
                raise ValueError("User not found\n")

        if role == "LS":
            student = Student(name, idNumber, contactNumber)
            student_record_management = StudentRecordManagement()
            found = student_record_management.findRecord(student.name,student.idNumber,student.contactNumber)
            if found:
                return True
            else:
                raise ValueError("User not found\n")

        if role == "NA":
           return True

        return False

File: assignment-5-Jwizzed-main/Authentication.cs
namespace ConsoleApplication1;

public class Authentication
{
    private Admin admin;
    private Student student;
    private AdminRecordManagement admin_record_management;
    private StudentRecordManagement student_record_management;
    public AdminRecordManagement getAdminRecordManagement()
    {
        return admin_record_management;
    }
    public StudentRecordManagement getStudentRecordManagement()
    {
        return student_record_management;
    }
    public Authentication()
    {
    }
    public void register(string name, int idNumber, string contactNumber, string role)
    {
        switch (role)
        {
            case "RA":
                admin = new Admin(name, idNumber, contactNumber);
                admin_record_management = new AdminRecordManagement();
                admin_record_management.add(admin);
                break;
            case "RS":
                student = new Student(name, idNumber, contactNumber);
                student_record_management = new StudentRecordManagement();
                student_record_management.add(student);
                break;
            default:
                throw new Exception("User not found");
        }
    } 
    public Boolean login(string name, int idNumber, string contactNumber, string role)
    {
        switch (role)
        {
            case "LA":
                admin = new Admin(name, idNumber, contactNumber);
                admin_record_management = new AdminRecordManagement();
                   if (admin_record_management.find(idNumber))
                   {
                       return true;
                   }
                   else 
                   {
                        throw new Exception("Login Failed");
                   }
            case "LS":
                student = new Student(name, idNumber, contactNumber);
                student_record_management = new StudentRecordManagement();
                   if (student_record_management.find(idNumber))
                   {
                       return true;
                   }
                   else 
                   {
                        throw new Exception("Login Failed");
                   }
            case "NA":
                return true;
            default:
                return false;
        }
    }
}